### DISCUSSION TITLE
Is pdm able to add dependencies in default group but with OS information?

### DISCUSSION THREAD

**Discussion by monchin** [pdm-project/pdm#2616] [Q&A] [answered]
Hello, I'm writing a small project with pdm. Now I want this project to be able to work across platform. 
In a function I need to set process affinity. When it is `posix`, I can directly use `os.sched_setaffinity`; but when it is `nt`, I must import `win32api`, `win32process` and `win32con` at first. 

As on Linux, `pypiwin32` can't be installed (pdm would say "pdm.exceptions.CandidateNotFound: No candidate is found for `pywin32` that matches the environment or hashes"), I'd like to know what is the best practice to deal with situations like this? Must I use different group like `pdm add -dG windows pypiwin32`? If other projects depend on this project, how to let them install `pypiwin32` when it is windows and not install `pypiwin32` when it is linux?

**Comment by frostming** [comment, marked as answer]
```
pdm add 'pypiwin32; sys_platform=="win32"'
```

https://peps.python.org/pep-0508/#environment-markers

**Reply by monchin** [reply]
Thank you! but I tried and found 
```
pdm add "pypiwin32; sys_platform=='win32'"
```
works!
